python/grass/jupyter/interactivemap.py

```diff
@@ -346,6 +347,95 @@ def add_layer_control(self, **kwargs):
         else:
             self.layer_control_object = self._ipyleaflet.LayersControl(**kwargs)
 
+    def setup_drawing_interface(self):
+        """
+        Allow Users to draw Geometry and Save/Delete them.
+        """
+        import ipywidgets as widgets  # pylint: disable=import-outside-toplevel
+
+        draw_control = self._ipyleaflet.DrawControl(edit=False, remove=False)
+        drawn_geometries = []
+        geo_json_layers = {}
+        save_button_control = None
+
+        def handle_draw(_target, action, geo_json):
+            if action == "created":
+                drawn_geometries.append(geo_json)
+                print(f"Geometry created: {geo_json}")
+
+        draw_control.on_draw(handle_draw)
+
+        draw_button = widgets.ToggleButton(
+            icon="pencil",
+            value=False,
+            tooltip="Click to draw geometries",
+            layout=widgets.Layout(width="33px", margin="0px 0px 0px 0px"),
+        )
+
+        def toggle_geometry(change):
+            if change["new"]:
+                self.map.add_control(draw_control)
+                show_interface()
+            else:
+                drawn_geometries.clear()
+                self.map.remove_control(draw_control)
+                hide_interface()
+
+        draw_button.observe(toggle_geometry, names="value")
+
+        def show_interface():
+            nonlocal save_button_control
+
+            name_input = widgets.Text(
+                description="Name:",
```

Comment:
```suggestion
                description="New vector map name:",
```

Reply:
This takes half of the space of the displayed widget. I have added an icon for the save button. Should I move the save button, or does the icon work?